worker/httpserver/manifold.go

```diff
@@ -159,55 +180,7 @@ func (config ManifoldConfig) start(context dependency.Context) (_ worker.Worker,
 		Mux:                  mux,
 	})
 	if err != nil {
-		close(abort)
 		return nil, errors.Trace(err)
 	}
-	return &wrapperWorker{
-		Worker:        w,
-		mux:           mux,
-		authenticator: authenticator,
-		cleanup: func() {
-			close(abort)
-			stTracker.Done()
-		},
-	}, nil
-}
-
-var (
-	muxType           = reflect.TypeOf(&apiserverhttp.Mux{})
-	authenticatorType = reflect.TypeOf((*httpcontext.LocalMacaroonAuthenticator)(nil)).Elem()
-)
-
-func manifoldOutput(in worker.Worker, out interface{}) error {
-	w, ok := in.(*wrapperWorker)
-	if !ok {
-		return errors.Errorf("expected worker of type %T, got %T", w, in)
-	}
-	rv := reflect.ValueOf(out)
-	if rt := rv.Type(); rt.Kind() == reflect.Ptr {
-		elemType := rt.Elem()
-		switch {
-		case muxType.AssignableTo(elemType):
-			rv.Elem().Set(reflect.ValueOf(w.mux))
-			return nil
-		case authenticatorType.AssignableTo(elemType):
-			rv.Elem().Set(reflect.ValueOf(w.authenticator))
-			return nil
-		}
-	}
-	return errors.Errorf("unexpected output type %T", out)
-}
-
-type wrapperWorker struct {
-	worker.Worker
-	mux           *apiserverhttp.Mux
-	authenticator httpcontext.LocalMacaroonAuthenticator
-	cleanupOnce   sync.Once
-	cleanup       func()
-}
-
-func (w *wrapperWorker) Wait() error {
-	err := w.Worker.Wait()
-	w.cleanupOnce.Do(w.cleanup)
-	return err
+	return common.NewCleanupWorker(w, func() { stTracker.Done() }), nil
```

Comment:
If we don't need the tracker, this gets simpler.

Reply:
:(